python/ray/tune/experiment/trial.py

```diff
@@ -1070,9 +1083,18 @@ def should_checkpoint(self):
         )
 
     def has_checkpoint(self):
+        if _use_storage_context():
+            return self.checkpoint is not None
+
         return self.checkpoint.dir_or_data is not None
 
     def clear_checkpoint(self):
+        if _use_storage_context():
+            if self.latest_checkpoint_result:
+                self.latest_checkpoint_result.checkpoint = None
+            self.temporary_state.restoring_from = None
+            self.run_metadata.invalidate_cache()
+
```

Comment:
Is this missing a return?

```suggestion
    return

```